app/models/infra_conversion_job.rb

```diff
@@ -489,6 +496,22 @@ def restore_vm_attributes
     # Save destination_vm in VMDB
     destination_vm.save
 
+    update_migration_task_progress(:on_exit)
+    queue_signal(:power_on_vm)
+  rescue StandardError
+    update_migration_task_progress(:on_error)
+    queue_signal(:power_on_vm)
+  end
+
+  def power_on_vm
+    update_migration_task_progress(:on_entry)
+
+    unless target_vm.power_state == 'on' || migration_task.options[:source_vm_power_state] != 'on'
```

Comment:
This inverted logic hurts my head when used with an OR.

Reply:
Agreed. I inverted it into if + and.